pkg/reconciler/cluster/cluster_controller.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"time"
+
+	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	apiresourceinformer "github.com/kcp-dev/kcp/pkg/client/informers/externalversions/apiresource/v1alpha1"
+	workloadinformer "github.com/kcp-dev/kcp/pkg/client/informers/externalversions/workload/v1alpha1"
+)
+
+func NewController(
```

Comment:
be specific what this does: NewHeartbeatWatcherController or so

Reply:
With the new package name is `workloadclusterheartbeat.NewController`, which I think is descriptive enough without being repetitive. Let me know if you disagree.